source/common/filter/tcp_proxy.cc

```diff
@@ -120,6 +120,8 @@ TcpProxy::TcpProxy(TcpProxyConfigSharedPtr config, Upstream::ClusterManager& clu
       upstream_callbacks_(new UpstreamCallbacks(this)) {}
 
 TcpProxy::~TcpProxy() {
+  request_info_.finalize(std::chrono::steady_clock::now());
```

Comment:
Hmmm, this is interesting.  I was going to ask that you make sure all the new fields are tracked for TCP proxy session too, but I see that we're already not consistently doing that.    @ggreenway given the logical things (first byte read, last byte sent) make sense for TCP proxy, but the request/response naming does not, do you think  we should do neutral naming (nix request/response) and track it here, neutral naming and track things for TCP proxy some day, or just comment that it's HTTP only?

Reply:
I think all the names in the new interfaces are in terms of upstream/downstream bytes, which should be fine.  The interface comments mention request/response, so that could be cleaned up.

I think in the tcpproxy, we should think about what makes sense.  For instance, firstDownstreamRx and firstUpstreamTx should always be the same time, so does it make sense to track both?  And for last byte in either direction, we don't know when it happens until the connection closes, so we'd have to take a timestamp on every single rx/tx, and use the last one after the connection closes.  

I think my preference is to sort out all those details in a separate PR.  This one maintains parity in tcp_proxy with previous state, which I this is good enough for this PR.